### PR TITLE
Fullstack integration - Docker postgreSQL container, NestJS, and React

### LINKED ISSUE #6: update all nest endpoints to have `/api/...` prefix
from what i understand our nest server should serve the react app for any request received except those that have /api in them
any request without /api should serve the react app and from there react-router will handle/interrupt the url changes therefore making it a SPA

### PR DESCRIPTION
**Pull request by jvendettuoli**
Summary: 
- Created a docker-compose yml file and associated package.json script that will start up a postgres container. 
- Updated backend to access this container (or any postgres database provided by .env file variables)
- Updated backend to accept incoming requests from frontend with `/api` prefix to all backend routes
  - Resolves #6 /api prefix issue
- There may be some style changes to Login component, but those are temporary. Login component will be on separate PR
- Minor changes because my editor autoformatted on save

PostgreSQL Docker Container (Optional):
- Requires docker and docker-compose to be installed locally. Since this isn't required to run app, I didn't make them required in package.json.
- To install [Docker](https://docs.docker.com/get-docker/)
  - I installed it on Linux, didn't run into any issues.
  - I did change my local setup to not require the use of sudo before docker commands. Don't anticipate this would be an issue for the run script.
- To install [Docker-Compose](https://docs.docker.com/compose/install/)
  - Again, no problem or additional setup on Linux
- Environmental variables required:
  - `DATABASE_PORT`: I used default postgres port 5432 
  - `DATABASE_USER`: your choice 
  - `DATABASE_PASSWORD`: your choice 
  - `DATABASE_DB`: name of postgres db, your choice
  - `DATABASE_HOST`: May not be required, I left it out and it worked, but I believe defaults to localhost
- When installed, the docker container can be created from web-backend directory via `npm run start:dev:db` which runs the docker-compose.yml file.
- Terminal should show docker container was created, but you can confirm with `docker ps` which shows running containers. Should be one that matches the name of `container_name` from docker-compose.yml file, in this case `postgres`.

Confirm all changes work:
- Either start postgreSQL docker container or have a local db with proper .env file in web-backend directory
- `npm run start:dev` from web-backend directory -> terminal should show no errors and display that user table is created in postgres database
- `cd client` from web-backend directory
- `npm start` in client directory -> should open webpage with no issues, showing latest styles from James
- navigate to `/login` via address bar or clicking LOGIN button
- click Sign In button - > network tab should show that a successful call was made to `/api`
- Make a POST request to `/api/users` with JSON body of `{"first_name": "TestFirst", "last_name": "TestLast"}` -> Responds with 404 and error from server, but if you `SELECT * FROM users` in the `postgres` database, it will have a new entry matching the sent data.
  - I used Insomnia to make the request, but anyway should work.

I'll have a follow up PR soon where I clean up the backend routes. This was just getting a bit large, and the connection works even if there isn't much of a useful response.

### CODE REVIEW

**Review by jd2rogers2 — APPROVED** [review]
😢 my trailing commas
i'd say update the readme file to have accurate set up and run instructions
also, if it's ok with you, i would just clone this repo down to your local instead of forking (or you could `git remote set-url origin https://github.com/Nonprofit-Exchange-Hub/web-backend.git`). i had to setup a remote to your repo and then pull in order to checkout your branch
would love to hear how you found all this stuff and walk me through your docker env (i'm still using locally running pg)
side note: /api/user endpoint didn't work for me when i used [these steps](https://github.com/Nonprofit-Exchange-Hub/web-backend/pull/7) (but add `/api` to those requests) but at least the BE builds now so A+

File: client/src/FilterGroup.tsx [hunk 1]
```diff
@@ -1,46 +1,40 @@
+import * as React from 'react';
 import { makeStyles } from '@material-ui/core/styles';
 import Checkbox from '@material-ui/core/Checkbox';
 import FormControlLabel from '@material-ui/core/FormControlLabel';
 import Typography from '@material-ui/core/Typography';
 
-import type { Theme } from '@material-ui/core/styles';
-
+import { Theme } from '@material-ui/core/styles';
```
[the remark this answers is not in the document]
Comment (jvendettuoli) [reply client/src/FilterGroup.tsx:7]: I'm not sure why, but my vscode was throwing an error when I tried to save that file after adding the React import I'll put it back in and see if I can figure out why it was upset.

File: client/src/Login.tsx [hunk 2]
```diff
@@ -69,66 +107,94 @@ function Login() {
         setShowPassword(!showPassword);
     };
 
-    const handleMouseDownPassword = (evt: React.MouseEvent) => {
+    const handleSubmit = async (evt: React.FormEvent) => {
         evt.preventDefault();
+        console.debug('handleSubmit - formData: ', formData);
+        try {
+            const res = await window.fetch('http://localhost:3001/api');
```
[the remark this answers is not in the document]
Comment (jvendettuoli) [reply client/src/Login.tsx:114]: Correct,  working on putting some test data together to actually have a function test of the login.

File: client/src/Login.tsx [hunk 3]
```diff
@@ -69,66 +107,94 @@ function Login() {
...
+        console.debug('googleSignIn - evt.currentTarget:', evt.currentTarget);
+    };
+
+    const facebookSignIn = (evt: React.MouseEvent) => {
+        console.debug('facebookSignIn - evt.currentTarget:', evt.currentTarget);
     };
 
     return (
-        <div className="Login">
+        <div className="Login" style={{ justifyContent: 'center', alignItems: 'center' }}>
```
[the remark this answers is not in the document]
Comment (jvendettuoli) [reply client/src/Login.tsx:130]: Nope, I was just testing it out. I'll get it in classes for the login-component PR.

File: package.json [hunk 4]
```diff
@@ -1,87 +1,83 @@
...
-		],
-		"coverageDirectory": "../coverage",
-		"testEnvironment": "node"
-	}
+    "name": "neh-backend",
+    "version": "0.0.1",
+    "description": "",
+    "author": "",
+    "private": true,
+    "license": "UNLICENSED",
```
[the remark this answers is not in the document]
Comment (jvendettuoli) [reply package.json:7]: Can do!

**Comment by jvendettuoli** [comment]
Thanks for reviewing James, I'll make the suggested changes outside of the Login component ones, which will be in a separate PR. I'll also take your suggestion for cloning the repo.
Before merging I will:
1. Figure out why the steps from your Jwt authentication PR aren't working.
2. Add MIT license
3. Figure out why I was having issue with `type {Theme}`
4. Update README once it works with the JWT steps.

**Pull request merged**